test/src/unit-cppapi-group.cc

```diff
@@ -1021,67 +1021,6 @@ class GroupDtorDoesntThrowException : public std::exception {
   }
 };
 
-TEST_CASE("C++ API: Group close group with error", "[cppapi][group][error]") {
```

Comment:
I know that's still a draft, but so that I don't forget to ask, why do we need to remove this test? could we just modify it to test what it wants to test?

Reply:
There's an important nuance here: this test is very specifically checking that the C++ API destructor doesn't throw (when calling `Group::close`). From my understanding, your changes fix the underlying issue, and the remote path will only throw if there's no `RestClient`, so it seems safe to remove this test (rather than modifying). My only nit is that you should remove the definition of  `GroupDtorDoesntThrowException` as well.

Reply:
Removed unused class.